tools/serverpod_cli/lib/src/util/server_directory_finder.dart

```diff
@@ -321,9 +321,22 @@ class ServerDirectoryFinder {
       }
       var homeDir =
           Platform.environment['HOME'] ?? Platform.environment['USERPROFILE'];
-      if (homeDir != null && p.normalize(dir.path) == p.normalize(homeDir)) {
+
+      final parentPath = p.canonicalize(dir.parent.resolveSymbolicLinksSync());
```

Comment:
I would suggest doing `final dirPath = dir. resolveSymbolicLinksSync()` very early (just after entry), and then don't use `dir.path` but `dirPath` throughout the method.

Also, I think you should _resolve_ `homeDir` as well. Might be 8.3 file alias as well.